src/resources/projects/controller.js

```diff
@@ -16,7 +18,28 @@ const getAllProjects = async (req, res) => {
         },
         donations: true,
       },
-    });
+    };
+
+    if (title) {
+      queryOptions.where = {
+        title: {
+          contains: title,
+        },
+      };
+    }
+
+    if (userName) {
+      queryOptions.where = {
```

Comment:
Nice dynamic query building. What if you want to query by both title *and* username? Right now this will only let you do one or the other

Reply:
This part suppose to work with search field on the frontend, to be able to search for projects either by the author name or the project title. Should I still update the backend to be able to query by both title and username?

Reply:
If that's how you want it to function, go with it. You can always change it to work with both later if you want to :)